lib/puppet/provider/katello_ssl_tool.rb

```diff
@@ -114,41 +59,20 @@ def common_args
        '--cert-expiration', resource[:expiration]]
     end
 
-    def rpmfile
-      path = self.build_path("#{rpmfile_base_name}")
-      path = path + "-[0-9].*" + "noarch.rpm"
-
-      rpmfile = Dir[path].max_by do |file|
-        version_from_name(file)
+    def pubkey(cert_name = resource[:name])
+      if resource.to_hash.key?(:hostname)
+        "#{resource[:build_dir]}/#{resource[:hostname]}/#{cert_name}.crt"
+      else
+        "#{resource[:build_dir]}/#{cert_name}.crt"
       end
```

Comment:
shouldn't this be using `build_path` (which, ironically, doesn't use `:build_dir`)?

Reply:
because then you don't need the `if hostname` branching, as `lib/puppet/provider/cert/katello_ssl_tool.rb` already does that:
https://github.com/theforeman/puppet-certs/blob/d7e8d4e981036ef5d237f708a5a5b48d69edc470/lib/puppet/provider/cert/katello_ssl_tool.rb#L53-L55

Reply:
Cleaned this whole area up to be more succinct.